src/wp-includes/functions.php

```diff
@@ -4947,14 +4947,31 @@ function _wp_array_get( $input_array, $path, $default_value = null ) {
 	}
 
 	foreach ( $path as $path_element ) {
-		if (
-			! is_array( $input_array ) ||
-			( ! is_string( $path_element ) && ! is_integer( $path_element ) && ! is_null( $path_element ) ) ||
-			! array_key_exists( $path_element, $input_array )
-		) {
+		if ( ! is_array( $input_array ) ) {
 			return $default_value;
 		}
-		$input_array = $input_array[ $path_element ];
+
+		if ( is_string( $path_element )
+			|| is_integer( $path_element )
+			|| null === $path_element
+		) {
+			/*
+			 * Check if the path element exists in the input array.
+			 * We check with `isset()` first as it is a lot faster
+			 * than `array_key_exists()`.
+			 * If `isset()` returns false, we check with `array_key_exists()`
+			 * which also checks for `null` values.
+			 */
+			if ( isset( $input_array[ $path_element ] ) ) {
```

Comment:
We can maybe just merge this condition and the `array_key_exists()` with an `||` here rather than repeating the assignment and `continue`.

As this slows down `null` detection, is there enough of a performance benefit to non-null to make this worthwhile on the average page load?

Reply:
That's what I initially had, but in my tests separating these 2 was significantly faster 🤷 Check out https://3v4l.org/kiAV9 for an example.

On average the function now runs 10%+ faster on my tests

Reply:
But after some time we definitely get new ticket for remove duplicate code. 🤔